ams/CleanupCommand.cs

```diff
@@ -0,0 +1,233 @@
+                {
+                    if (_options.IsCleanUpAccount)
+                    {
+                        var endpoints = account.GetStreamingEndpoints();
+                        var policies = account.GetContentKeyPolicies();
+                        var liveevents = account.GetMediaLiveEvents();
+                        var locator = await account.GetStreamingLocatorAsync(asset, cancellationToken);
+                        await CleanUpContentAsync(container, asset, locator, endpoints, policies, liveevents);
+                        _logger.LogDebug("account {account} is deleted.", account.Data.Name);
+
+                    }
+                    else
+                    {
+                        var locator = account.GetStreamingLocatorAsync(asset, cancellationToken).Result;
+                        await CleanUpContentAsync(container, asset, locator, null, null, null);
+                        _logger.LogDebug("locator: {locator}, Migrated asset: {asset} , container: {container} are deleted.", locator?.Data.Name, asset.Data.Name, container?.Name);
+                    }
+                        return true;
+                }
+                else
+                {
+                    _logger.LogDebug("asset: {asset} with migration status: {migrateResult} does not meet the criteria for deletion.", asset.Data.Name, migrateResult.Status);
+                    return false;
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to delete asset {name}", asset.Data.Name);
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Delete the resources that were created.
+        /// </summary>
+        /// <param name="inputAssets">The input Asset List.</param>
+        /// <param name="streamingLocator">The streaming locator. </param>
+        /// <returns>A task.</returns>
+        private async Task CleanUpContentAsync(
+            BlobContainerClient container,
+            MediaAssetResource? inputAsset,
+            StreamingLocatorResource? streamingLocator,
+            StreamingEndpointCollection? streamingEndpoints,
+            ContentKeyPolicyCollection? contentKeyPolicies,
+            MediaLiveEventCollection? liveEvents)
+        {
+            
+            if (streamingLocator != null)
+            {
+                await streamingLocator.DeleteAsync(WaitUntil.Completed);
+            }
+
+            if (inputAsset != null)
+            {
+                await inputAsset.DeleteAsync(WaitUntil.Completed);
+            }
+
+            if (streamingEndpoints != null)
+            {
+                foreach (var streamingEndpoint in streamingEndpoints)
+                {
+                    streamingEndpoint.DeleteAsync(WaitUntil.Completed);
```

Comment:
await is required.

For StreamingEndpoints and LiveEvents, before they can be deleted, the code would need to stop them first if they are in running state.

Reply:
updated